dashboard/api.py

```diff
@@ -0,0 +1,227 @@
+"""
+Apis for the dashboard
+"""
+import datetime
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class CourseStatus(object):
+    """
+    Possible statuses for a course for a user
+    """
+    PASSED = 'passed'
+    NOT_PASSED = 'not-passed'
+    CURRENT_GRADE = 'verified-not-completed'
+    UPGRADE = 'enrolled-not-verified'
+    NOT_OFFERED = 'not-offered'
+    OFFERED = 'offered-not-enrolled'
+
+    @classmethod
+    def all_statuses(cls):
+        """Helper to get all the statuses"""
+        return [cls.PASSED, cls.NOT_PASSED, cls.CURRENT_GRADE,
+                cls.UPGRADE, cls.NOT_OFFERED, cls.OFFERED]
+
+
+class CourseRunStatus(object):
+    """
+    Possible statuses for a course run for a user
+    """
+    NOT_ENROLLED = 'not-enrolled'
+    GRADE = 'grade'
+    READ_CERT = 'read-cert'
+    WILL_ATTEND = 'will-attend'
+    UPGRADE = 'upgrade'
+    NOT_PASSED = 'not-passed'
+
+
+class CourseFormatConditionalFields(object):
+    """
+    The formatting of a course run is dependent
+    on the CourseStatus status passed on the function.
+    There are some fields that are common and others
+    that depend on the status. Also the name of the fields changes.
+
+    This class contains the association between the CourseStatus status
+    that need specific fields, the field associated correspondent
+    to a course run and the new name they need to have.
+    """
+    ASSOCIATED_FIELDS = {
+        CourseStatus.OFFERED: [
+            {
+                'course_run_field': 'enrollment_start',
+                'format_field': 'enrollment_start_date'
+            },
+            {
+                'course_run_field': 'fuzzy_enrollment_start_date',
+                'format_field': 'fuzzy_enrollment_start_date'
+            },
+        ],
+        CourseStatus.CURRENT_GRADE: [
+            {
+                'course_run_field': 'start_date',
+                'format_field': 'course_start_date'
+            },
+        ]
+    }
+
+    @classmethod
+    def get_assoc_field(cls, course_status):
+        """
+        Method to get from the ASSOCIATED_FIELDS dict
+        """
+        return cls.ASSOCIATED_FIELDS.get(course_status, [])
+
+
+class CourseRunUserStatus(object):
+    """
+    Representation of a course run status for a specific user
+    """
+    def __init__(self, status, course_run=None, enrollment_for_course=None):
+        self.status = status
+        self.course_run = course_run
+        self.enrollment_for_course = enrollment_for_course
+
+
+def get_info_for_course(user, course, user_enrollments, user_certificates):
```

Comment:
My IDE says that `user_certificates` is unused here. Will this be used in the future?